modin/pandas/dataframe.py

```diff
@@ -819,12 +819,57 @@ def explode(
             pandas.DataFrame.explode, column, ignore_index=ignore_index
         )
 
+    def _update_var_dicts_in_kwargs(self, expr, **kwargs):
+        """
+        Copy variables with "@" prefix in `local_dict` and `global_dict`.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to search variables with "@" prefix.
+        **kwargs : dict
+            See the documentation for eval() for complete details on the keyword arguments accepted by query().
+
+        Returns
+        -------
+        kwargs : dict
+            Dict with updated `local_dict` and `global_dict` keys.
+        """
+        import re
+
+        frame = sys._getframe()
+        try:
+            f_locals = frame.f_back.f_back.f_locals
+            f_globals = frame.f_back.f_back.f_globals
+        finally:
+            del frame
+        local_names = re.findall(r"@([\w]*)", expr)
```

Comment:
```suggestion
        local_names = set(re.findall(r"@([\w]+)", expr))
```
to remove duplicates to speed up iteration and to require at least one symbol in the name :) (otherwise it would pick empty string in e.g. `" hey @ someone!"`)